frontend/lib/i18n/i18n.tsx

```diff
@@ -0,0 +1,64 @@
+import { Fragment, ReactElement } from "react";
+
+import { locale, translations } from ".";
+
+import { Translation } from "./i18n.types";
+
+export type RenderCallback = (text: string) => ReactElement;
+
+export function translate(k: keyof Translation): string {
+  return translations[locale][k];
+}
+
+export function t(k: keyof Translation, vars?: Record<string, string>): string {
+  if (vars) {
+    return Object.entries(vars).reduce(
+      (acc, [key, value]) => (typeof value === "string" ? acc.replace(`$${key}`, value) : acc),
+      translate(k),
+    );
+  }
+
+  return translate(k);
+}
+
+export function tx(k: keyof Translation, vars?: Record<string, RenderCallback | string>): Array<string | ReactElement> {
```

Comment:
Can you add some JSDoc to this function? The parameter names `k` and `vars` are not really speaking to me.